### PR TITLE
Difference - Ekaterina – Sunday class

### LINKED ISSUE #12: Difference: filter (sunday class solution)

### PR DESCRIPTION
**Pull request by katsmamina**
Before the reviewer merges this PR to main/master, all of these boxes must be checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [x] The documentation is rendered into the README
  - [x] Docs were generated just before pushing
  - [x] Docs in the README matches the JSDoc comment
  - [x] The README contains at least one use case
- **Unit Tests**
  - [x] Unit tests are present
  - [x] There are at least 8 test cases
  - [x] All `describe`s and `it`s have clear and helpful names
  - [x] All test names actually describe what is being tested
  - [x] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [ ] Side-effects are tested, if necessary
  - [x] Tests were run just before pushing
  - [x] All tests pass
  - [x] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [ ] Variables have names that help understand the strategy
  - [ ] Spacing and formatting help to understand the strategy
  - [ ] The code is written as simply and clearly as possible
  - [ ] Use short, clear comments to describe the strategy
- **README**
  - [ ] There is an explanation of the solution's strategy in the README
  - [ ] The explanation is clear and understandable
  - [ ] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] The code was formatted just before pushing
  - [x] No errors occurred while formatting the code
- **Linting**
  - [x] There should be no linting errors or warnings
  - [x] Spelling should be correct in all code, comments and docstrings
  - [x] The code was linted just before pushing
- **The solution file**
  - [ ] Has a JSDoc comment
  - [ ] Has an exported function
  - [ ] Does no other code or comments outside of the function body (you can use `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [ ] Be well and consistently formatted
  - [ ] Have clear and helpful variable names
  - [ ] Declare variables in the smallest scope possible
  - [ ] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [ ] Have extra comments that are not about the solution's strategy
  - [ ] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [ ] Have any `console` calls (`.log`, `.assert`, ...)
  - [ ] Have any `debugger` statements

### CODE REVIEW

File: src/difference/solution-class-sunday/difference.spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  it('mixed arrays of strings, numbers, booleans', () => {
+    const actual = difference(
+      ['hello', true, false, 0, 1, 2],
+      ['hello', false, 1, 2],
+    );
+    expect(actual).toEqual([true, 0]);
+  });
+});
+
+describe('checks arrays which include repeating elements', () => {
```
Comment (hserdogan94) [inline src/difference/solution-class-sunday/difference.spec.js:21]: I think we should remove duplicated values in the returned array like in the example in line 17 src/difference/solution-class-sunday/difference.js

File: src/difference/solution-class-sunday/difference.spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  it('arrays of strings including repeating', () => {
+    const actual = difference(
+      ['hello', 'hello', 'thank you', 'goodbye'],
+      ['hi', 'hi', 'thank you', 'bye'],
+    );
+    expect(actual).toEqual(['hello', 'hello', 'goodbye']);
+  });
+  it('arrays of numbers including repeating', () => {
```
Comment (hserdogan94) [inline src/difference/solution-class-sunday/difference.spec.js:29]: For example we expect that as a returned value = [3]

File: src/difference/solution-class-sunday/difference.spec.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { difference } from './difference.js';
```
Comment (hserdogan94) [inline src/difference/solution-class-sunday/difference.spec.js:1]: And i think we should add tests for returning new array and original one is not modified

File: src/difference/solution-class-sunday/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+## Implementation
+
+1. Create the criteria for filtering – values not included into first array
+2. Create a new array with using the ‘filter’ array method
+
+---
+
+## Use Cases
```
Comment (hserdogan94) [inline src/difference/solution-class-sunday/README.md:26]: Very good case example :)

File: src/difference/solution-class-sunday/difference.spec.js (hunk 1 above)
Comment (katsmamina) [reply src/difference/solution-class-sunday/difference.spec.js:21]: all tests are working! and it was necessary to check what will happen with such arrays that include duplicated values

Comment (hserdogan94) [reply src/difference/solution-class-sunday/difference.spec.js:21]: Yes but

```
**Example**

```js
difference([1, 2, 1], [2, 3]); //  [1]
```
like in the example it should return [1] not [1,1]
for this that link can be helpful for you 

Removing duplicates;
https://levelup.gitconnected.com/7-ways-to-remove-duplicates-from-array-in-javascript-cea4144caf31

 if you wish to change; otherwise, everything others seem good and clear, readable. If you would like to continue with other exercises I will merge this. Please let me know. Good luck.
